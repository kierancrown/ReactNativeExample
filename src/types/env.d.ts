declare module '@env' {
  export const DARKSKY_API_KEY: string;
  export const NEWS_API_KEY: string;
}
