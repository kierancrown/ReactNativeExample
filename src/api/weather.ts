import {DARKSKY_API_KEY} from '@env';
import axios from 'axios';

const getWeatherForecast = async (lat: number, long: number) => {
  try {
    console.log(
      `https://api.darksky.net/forecast/${DARKSKY_API_KEY}/${lat},${long}?units=uk2`,
    );
    const res = await axios.get(
      `https://api.darksky.net/forecast/${DARKSKY_API_KEY}/${lat},${long}?units=uk2`,
    );
    if (res.status === 200) {
      return res.data;
    }
  } catch (error) {
    console.error('Error getting weather forecast', error);
  }
};

export {getWeatherForecast};
