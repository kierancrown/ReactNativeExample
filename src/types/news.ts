export module News {
  export interface Source {
    id: string;
    name: string;
  }

  export interface Article {
    source: Source;
    author: string;
    title: string;
    description: string;
    url: string;
    urlToImage: string;
    publishedAt: Date;
    content: string;
  }

  export interface HeadlinesResponse {
    status: string;
    totalResults: number;
    articles: Article[];
  }
}

export interface ParseResult {
  title: string | null;
  content: string | null;
  author: string | null;
  date_published: string | null;
  lead_image_url: string | null;
  dek: string | null;
  next_page_url: string | null;
  url: string;
  domain: string;
  excerpt: string | null;
  word_count: number;
  direction: 'ltr' | 'rtl';
  total_pages: number;
  rendered_pages: number;
}
