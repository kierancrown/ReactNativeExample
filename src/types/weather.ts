export module Weather {
  export interface Currently {
    time: number;
    summary: string;
    icon: string;
    nearestStormDistance: number;
    nearestStormBearing: number;
    precipIntensity: number;
    precipIntensityError: number;
    precipProbability: number;
    precipType: string;
    temperature: number;
    apparentTemperature: number;
    dewPoint: number;
    humidity: number;
    pressure: number;
    windSpeed: number;
    windGust: number;
    windBearing: number;
    cloudCover: number;
    uvIndex: number;
    visibility: number;
    ozone: number;
  }

  export interface Datum {
    time: number;
    precipIntensity: number;
    precipIntensityError: number;
    precipProbability: number;
    precipType: string;
  }

  export interface Minutely {
    summary: string;
    icon: string;
    data: Datum[];
  }

  export interface Datum2 {
    time: number;
    summary: string;
    icon: string;
    precipIntensity: number;
    precipProbability: number;
    precipType: string;
    temperature: number;
    apparentTemperature: number;
    dewPoint: number;
    humidity: number;
    pressure: number;
    windSpeed: number;
    windGust: number;
    windBearing: number;
    cloudCover: number;
    uvIndex: number;
    visibility: number;
    ozone: number;
  }

  export interface Hourly {
    summary: string;
    icon: string;
    data: Datum2[];
  }

  export interface Datum3 {
    time: number;
    summary: string;
    icon: string;
    sunriseTime: number;
    sunsetTime: number;
    moonPhase: number;
    precipIntensity: number;
    precipIntensityMax: number;
    precipIntensityMaxTime: number;
    precipProbability: number;
    precipType: string;
    temperatureHigh: number;
    temperatureHighTime: number;
    temperatureLow: number;
    temperatureLowTime: number;
    apparentTemperatureHigh: number;
    apparentTemperatureHighTime: number;
    apparentTemperatureLow: number;
    apparentTemperatureLowTime: number;
    dewPoint: number;
    humidity: number;
    pressure: number;
    windSpeed: number;
    windGust: number;
    windGustTime: number;
    windBearing: number;
    cloudCover: number;
    uvIndex: number;
    uvIndexTime: number;
    visibility: number;
    ozone: number;
    temperatureMin: number;
    temperatureMinTime: number;
    temperatureMax: number;
    temperatureMaxTime: number;
    apparentTemperatureMin: number;
    apparentTemperatureMinTime: number;
    apparentTemperatureMax: number;
    apparentTemperatureMaxTime: number;
  }

  export interface Daily {
    summary: string;
    icon: string;
    data: Datum3[];
  }

  export interface Alert {
    title: string;
    regions: string[];
    severity: string;
    time: number;
    expires: number;
    description: string;
    uri: string;
  }

  export interface Flags {
    sources: string[];
    'nearest-station': number;
    units: string;
  }

  export interface RootObject {
    latitude: number;
    longitude: number;
    timezone: string;
    currently: Weather.Currently;
    minutely: Weather.Minutely;
    hourly: Weather.Hourly;
    daily: Weather.Daily;
    alerts: Weather.Alert[];
    flags: Weather.Flags;
    offset: number;
  }
}
