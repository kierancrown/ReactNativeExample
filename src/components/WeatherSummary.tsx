import React from 'react';
import {View, Text, StyleSheet, useColorScheme} from 'react-native';
import {Weather} from '../types/weather';

interface WeatherProps {
  weather: Weather.Currently | undefined;
}

const WeatherSummary = ({weather}: WeatherProps) => {
  const isDarkMode = useColorScheme() === 'dark';

  const styles = StyleSheet.create({
    container: {
      flexDirection: 'row',
      marginVertical: 16,
      marginTop: 21,
    },
    textContainer: {
      paddingLeft: 8,
    },
    temp: {
      fontSize: 32,
      fontWeight: 'bold',
      color: isDarkMode ? '#fff' : '#000',
    },
    condition: {
      fontSize: 16,
      fontWeight: '500',
      color: isDarkMode ? '#fff' : '#000',
    },
  });

  return (
    <View style={styles.container}>
      {/* Placeholder icon */}
      <View
        // eslint-disable-next-line react-native/no-inline-styles
        style={{
          width: undefined,
          height: 60,
          aspectRatio: 1,
          borderRadius: 30,
          backgroundColor: isDarkMode ? 'white' : 'black',
        }}
      />
      <View style={styles.textContainer}>
        {weather === undefined ? (
          <Text>Unable to fetch weather</Text>
        ) : (
          <>
            <Text style={styles.temp}>
              {Math.round(weather.temperature || 0)}Â°
            </Text>
            <Text style={styles.condition}>{weather.summary}</Text>
          </>
        )}
      </View>
    </View>
  );
};

export default WeatherSummary;
