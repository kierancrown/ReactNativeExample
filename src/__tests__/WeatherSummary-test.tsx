import * as React from 'react';
import WeatherSummary from '../components/WeatherSummary';
import renderer from 'react-test-renderer';
import {Weather} from '../types/weather';

const mockProps: Weather.Currently = {
  time: 1648494869,
  summary: 'Drizzle',
  icon: 'rain',
  nearestStormDistance: 0,
  precipIntensity: 0.1678,
  precipIntensityError: 0.0352,
  precipProbability: 0.99,
  precipType: 'rain',
  temperature: 12.2,
  apparentTemperature: 12.2,
  dewPoint: 7.82,
  humidity: 0.75,
  pressure: 1017.3,
  windSpeed: 5.37,
  windGust: 10.45,
  windBearing: 89,
  cloudCover: 0.91,
  uvIndex: 0,
  visibility: 10,
  ozone: 404.9,
  nearestStormBearing: 89,
};

it('renders correctly', () => {
  const tree = renderer.create(<WeatherSummary weather={mockProps} />);
  expect(tree).toMatchSnapshot();
});
