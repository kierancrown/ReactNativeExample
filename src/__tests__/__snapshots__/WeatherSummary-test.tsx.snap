// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 21,
      "marginVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "black",
        "borderRadius": 30,
        "height": 60,
        "width": undefined,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 32,
          "fontWeight": "bold",
        }
      }
    >
      12
      Â°
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
          "fontWeight": "500",
        }
      }
    >
      Drizzle
    </Text>
  </View>
</View>
`;
